introspection-engine/connectors/introspection-connector/src/lib.rs

```diff
@@ -84,3 +84,10 @@ pub struct IntrospectionContext {
     pub source: Datasource,
     pub preview_features: BitFlags<PreviewFeature>,
 }
+
+impl IntrospectionContext {
+    pub fn foreign_keys_enabled(&self) -> bool {
+        self.source.referential_integrity.uses_foreign_keys()
+            && self.preview_features.contains(PreviewFeature::ReferentialIntegrity)
+    }
```

Comment:
It's nitpicky, but I'd rather see it the other way, since it's only checked in a single place:

```
pub fn referential_integrity_is_prisma()
```

Reply:
Yeah, we need that negation which is not cool, but I also don't like the name of your function ;)

Reply:
It's mostly that we don't check it anywhere else, but we assume foreign keys are enabled.